src/main/kotlin/com/ajaxproject/telegrambot/bot/BankInfoBot.kt

```diff
@@ -1,36 +1,49 @@
 package com.ajaxproject.telegrambot.bot
 
-import com.ajaxproject.telegrambot.bot.commands.AbstractCommand
+import com.ajaxproject.telegrambot.bot.model.UserRequest
+import com.ajaxproject.telegrambot.bot.service.UserSessionService
 import org.slf4j.LoggerFactory
-import org.springframework.beans.factory.annotation.Qualifier
 import org.springframework.beans.factory.annotation.Value
 import org.springframework.stereotype.Component
 import org.telegram.telegrambots.bots.TelegramLongPollingBot
 import org.telegram.telegrambots.meta.api.objects.Update
-
 @Component
 class BankInfoBot(
-    @Qualifier("commandsMap")
-    private val commandsMapping: Map<String, AbstractCommand>,
+    val dispatcher: Dispatcher,
+    val userSessionService: UserSessionService,
 ) : TelegramLongPollingBot() {
-
-    @Value("\${credentials.botUsername}")
+    @Value("\${bot.username}")
     private val botUsername: String = ""
 
-    @Value("\${credentials.botToken}")
+    @Value("\${bot.token}")
     private val token: String = ""
+    override fun getBotToken(): String = token
 
     override fun getBotUsername(): String = botUsername
-    override fun getBotToken(): String = token
 
     override fun onUpdateReceived(update: Update) {
-        if (update.hasMessage()) {
-            val text = update.message.text
-            commandsMapping[text]?.handle(update, this)
-        } else if (update.hasCallbackQuery()) {
-            val callBack = update.callbackQuery.data
-            log.debug("New request with callbackQuery: {}", callBack)
-            commandsMapping[callBack]?.handle(update, this)
+        if (update.hasMessage() && update.message.hasText()) {
+            val textFromUser = update.message.text
+
+            val userId = update.message.from.id
+            val userFirstName = update.message.from.firstName
+
+            log.info("[{}, {}] : {}", userId, userFirstName, textFromUser)
+
+            val chatId = update.message.chatId
+            val session = userSessionService.getSession(chatId)
+
+            val userRequest = UserRequest(
+                update = update,
+                userSession = session,
+                chatId = chatId
+            )
+            val dispatcher = dispatcher.dispatch(userRequest)
+            if (!dispatcher) {
+                log.warn("Unexpected update from user")
+            }
+        } else {
+            log.warn("Unexpected update from user")
```

Comment:
suppose your bot has multiple users. Suppose a dozens of users faced with some issues. So, you open logs and see: `Unexpected update from user`. It doesn't help to troubleshoot the issue. 

What can we do to make the log messages more useful? We can add more context!
For example:
```kotlin
log.warn("Received unexpected update from user: userId={}, updateDetails={}", userId, <some additional data>)
```